scripts/vnc-ssh-tunnel.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+set -e
+
+show_help()
+{
+cat << EOF
+usage: $(basename $BASH_SOURCE) [-p|--ssh-port port] host vnc_port
+
+Tunnels the vnc connection through ssh.
+
+host: hostname
+vnc_port: vnc server port at the host
+
+Options:
+    -p|--ssh-port: ssh port at the host. Default 22
+    -h|--help: show help
+EOF
+    exit 1
+}
+
+find_free_port()
+{
+    local start=49152
+    local range=5000
+    local i=0
+    while true; do
+        local port=$[$start + ($i)]
+        (echo -n >/dev/tcp/127.0.0.1/${port}) >/dev/null 2>&1
+        if [ $? -ne 0 ]; then
+            echo $port
+            break
+        fi
+        ((i++))
+        if [ $i -ge 5000 ]; then
+            echo "0"
+            break
+        fi
+    done
+}
+
+main()
+{
+    TEMP=$(getopt -o 'hp:' --long 'help,ssh-port:' -n "$(basename $BASH_SOURCE)" -- "$@")
```

Comment:
Isn't option parsing overkill for this? Also, I believe that `TEMP` could really be a local.

Generally, I would just keep a shell script like this simple and just use environment variables for options. I.e. if you wanted a different port, you would have to call `SSH_PORT=2200 vnc-ssh-tunnel.sh`.